providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/EventStreamObserver.java

```diff
@@ -0,0 +1,101 @@
+package dev.openfeature.contrib.providers.flagd;
+
+import java.util.Map;
+import io.grpc.stub.StreamObserver;
+import dev.openfeature.flagd.grpc.Schema.EventStreamResponse;
+import dev.openfeature.sdk.ProviderEvaluation;
+import com.google.protobuf.Value;
+
+/**
+ * EventStreamObserver handles events emitted by flagd.
+ */
+public class EventStreamObserver implements StreamObserver<EventStreamResponse> {
+    private EventStreamCallback callback;
+    private Boolean cacheEnabled;
+    private Map<String, ProviderEvaluation<Boolean>> booleanCache;
+    private Map<String, ProviderEvaluation<String>> stringCache;
+    private Map<String, ProviderEvaluation<Double>> doubleCache;
+    private Map<String, ProviderEvaluation<Integer>> integerCache;
+    private Map<String, ProviderEvaluation<dev.openfeature.sdk.Value>> objectCache;
+
+
+    static final String CONFIGURATION_CHANGE = "configuration_change";
+    static final String PROVIDER_READY = "provider_ready";
+
+    EventStreamObserver(
+        Boolean cacheEnabled, Map<String, ProviderEvaluation<Boolean>> booleanCache,
+        Map<String, ProviderEvaluation<String>> stringCache, Map<String, ProviderEvaluation<Double>> doubleCache,
+        Map<String, ProviderEvaluation<Integer>> integerCache,
+        Map<String, ProviderEvaluation<dev.openfeature.sdk.Value>> objectCache, EventStreamCallback callback
+    ) {
+        this.cacheEnabled = cacheEnabled;
+        this.booleanCache = booleanCache;
+        this.stringCache = stringCache;
+        this.doubleCache = doubleCache;
+        this.integerCache = integerCache;
+        this.objectCache = objectCache;
+        this.callback = callback;
+    }
+
+    @Override
+    public void onNext(EventStreamResponse value) {
+        switch (value.getType()) {
+            case CONFIGURATION_CHANGE:
+                this.handleConfigurationChangeEvent(value);
+                break;
+            case PROVIDER_READY:
+                this.handleProviderReadyEvent();
+                break;
+            default:
+                // log
+                return;
+        }
+    }
+
+    @Override
+    public void onError(Throwable t) {
+        t.printStackTrace();
+        this.purgeCache();
+        this.callback.setEventStreamAlive(false);
+        try {
+            this.callback.restartEventStream();
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Let's log instead of `e.printStackTrace()`. Please check other such usages as well 

ex:- ` log.warn("Unknown event type: " +value.getType(), e)`

Reply:
Done, thanks